manifests/java.pp

```diff
@@ -29,49 +32,31 @@
   Stdlib::Absolutepath $java_keystore,
   Optional[String] $source = undef,
   Optional[Pattern['^[A-Za-z0-9+/\n=-]+$']] $content = undef,
+  Stdlib::Absolutepath $filename = "/tmp/${name}-trustedca",
 ) {
   if ! defined(Class['trusted_ca']) {
     fail('You must include the trusted_ca base class before using any trusted_ca defined resources')
   }
 
   if $source and $content {
     fail('You must not specify both $source and $content for trusted_ca defined resources')
-  }
-
-  if $source {
-    file { "/tmp/${name}-trustedca":
-      ensure => 'file',
-      source => $source,
-      notify => Exec["validate /tmp/${name}-trustedca"],
-      mode   => '0644',
-      owner  => 'root',
-      group  => 'root',
-    }
-  } elsif $content {
-    file { "/tmp/${name}-trustedca":
-      ensure  => 'file',
-      content => $content,
-      notify  => Exec["validate /tmp/${name}-trustedca"],
-      mode    => '0644',
-      owner   => 'root',
-      group   => 'root',
-    }
-  } else {
+  } elsif !$source and !$content {
     fail('You must specify either $source or $content for trusted_ca defined resources')
   }
 
-  # This makes sure the certificate is valid
-  exec { "validate /tmp/${name}-trustedca":
-    command     => "openssl x509 -in /tmp/${name}-trustedca -noout",
-    logoutput   => on_failure,
-    path        => $trusted_ca::path,
-    notify      => Exec["import /tmp/${name}-trustedca to jks ${java_keystore}"],
-    returns     => 0,
-    refreshonly => true,
+  file { $filename:
+    ensure       => 'file',
+    source       => $source,
+    content      => $content,
+    mode         => '0644',
+    owner        => 'root',
+    group        => 'root',
+    validate_cmd => '/usr/bin/openssl x509 -in %s -noout',
```

Comment:
are you sure that this path is correct on all operating systems? Is there something around that uses /bin/openssl? We had problems like that in the past.

Reply:
Pretty sure. At least on EL (7+) `/bin` is a symlink to `/usr/bin`. Debian and Ubuntu both have it in `/usr/bin`. Not sure about suse mentioned in `metadata.json`, but I'd be quite confident in assuming it.

I wasn't sure if you can simply use `openssl` and expect it to use `$path`.

Reply:
as far as I know it needs to be an absolute path, that's how we implemented it in other modules as well (puppet/nginx, saz/ssh)